source/server/internal_stats_handler.cc

```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+#include <map>
+#include <set>
+
+#include "envoy/server/internal_stats_handler.h"
+#include "envoy/stats/stats.h"
+#include "envoy/stats/store.h"
+
+namespace Envoy {
+namespace Server {
+
+void InternalStatsHandler::receiveGlobalStats(std::set<std::string>& stat_names,
+                                              PostReceiveCb cb) const {
+  // TODO(ramaraochavali): See the comment in ThreadLocalStoreImpl::histograms() for why we use a
+  // multimap here. This makes sure that duplicate histograms get output. When shared storage is
+  // implemented this can be switched back to a normal map.
+  std::multimap<std::string, const Envoy::Stats::HistogramStatistics&> all_histograms;
+  for (const Stats::ParentHistogramSharedPtr& histogram : store_root_.histograms()) {
+    auto iterator = stat_names.find(histogram->name());
+    if (iterator != stat_names.end()) {
+      const Stats::HistogramStatistics& hist = histogram->intervalStatistics();
+      all_histograms.emplace(histogram->name(), hist);
+    }
+  }
+  cb(all_histograms);
```

Comment:
Why do we need to use continuation-passing style here, rather than just returning back the map to the caller?

Reply:
Yeah, I think we can just return the map here. Was initially thinking that this might be useful in non main threads, but doesn't seem super relevant.